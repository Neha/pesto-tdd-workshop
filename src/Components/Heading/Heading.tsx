import React from 'react';

const Heading = () => {
    return( <>
    <h1 className="s-20">Hello</h1></>)
}

export default Heading;
