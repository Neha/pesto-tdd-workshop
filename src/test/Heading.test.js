import { render,screen } from "@testing-library/react";
import Heading from '../Components/Heading/Heading';

describe("Heading component" , () => {
    it("Heading should be H1",() => {
        render(<Heading/>);
        const headingElement = screen.getByRole('heading',{ level : 1});
        expect(headingElement).toBeInTheDocument();
    })
    it("Heading should have content",() => {
        render(<Heading/>);
        const headingContent = screen.getByRole('heading',{ name : "Hello"});
        expect(headingContent).toBeInTheDocument();
    })
    it("Heading should have classname s-20",() => {
        render(<Heading/>);
        const headingContent = screen.getByRole('heading');
        expect(headingContent).toHaveClass('s-20');
    })
})