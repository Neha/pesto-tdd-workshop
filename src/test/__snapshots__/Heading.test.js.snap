// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component Should match snapshot 1`] = `
<h1
  className="s-20"
>
  Hello
</h1>
`;
